sdks/java/core/src/main/java/org/apache/beam/sdk/options/ProxyInvocationHandler.java

```diff
@@ -552,7 +557,19 @@ public void serialize(PipelineOptions value, JsonGenerator jgen, SerializerProvi
         jgen.writeObject(serializableOptions);
 
         List<Map<String, Object>> serializedDisplayData = Lists.newArrayList();
-        for (DisplayData.Item<?> item : DisplayData.from(value).items()) {
+        DisplayData displayData;
+
+        try {
+          displayData = DisplayData.from(value);
+        } catch (Exception e) {
+          String msg = "Exception thrown while collecting display data for PipelineOptions. "
```

Comment:
I guess what I was wondering is if we could make sure that if there are 30 items, and we have trouble producing display data for 3 of them, we still get 27 items worth of display data.